### PR TITLE
Accept EPP-Template for Settings-File

### PR DESCRIPTION
**Pull request by cocker-cc**

### CODE REVIEW

**Comment by ekohl** [comment]
I wonder if we shouldn't move to EPP entirely. Thoughts?

**Comment by cocker-cc** [comment]
> I wonder if we shouldn't move to EPP entirely. Thoughts?

(+1)  EPP has the Advantage of accepting Sensitive natively, without the Need to unwrap.

File: manifests/settings_file.pp [hunk 1]
```diff
@@ -33,7 +33,13 @@
   if $ensure == 'absent' {
     $content = undef
   } else {
-    $content = template($template_path)
+    $content = if $template_path.match(/\.epp$/) {
+      epp($template_path)
```
Comment (ekohl) [inline manifests/settings_file.pp:37]: It'd be nice if we could pass in an explicit context as well.

**Review by ekohl — COMMENTED** [review]
I'd be happy if we move all the templates to EPP. It's probably a big refactor (we have a sizable number of templates in this module). This could be a good start.

File: manifests/settings_file.pp (hunk 1 above)
Comment (cocker-cc) [reply manifests/settings_file.pp:37]: What exactly do you mean by "context"?

Comment (ekohl) [reply manifests/settings_file.pp:37]: You can call `epp($template_path, $context)` so I'm thinking about a new parameter to the defined type.

Comment (cocker-cc) [reply manifests/settings_file.pp:37]: That`s fine by me, but in a different Pullrequest then.

**Review by ekohl — APPROVED** [review]
I thought about changing the data type to `Pattern[/\.(epp|erb)$/]` but I consider this mostly a private API and users can find out easily enough what works. Thanks!

**Pull request merged**